pytorch_lightning/accelerators/accelerator.py

```diff
@@ -102,11 +103,21 @@ def start_predicting(self, trainer: 'pl.Trainer') -> None:
 
     def pre_dispatch(self, trainer: 'pl.Trainer') -> None:
         """Hook to do something before the training/evaluation/prediction starts."""
+        self.move_optimizer_state()
+
         self.training_type_plugin.pre_dispatch()
         if self.training_type_plugin.setup_optimizers_in_pre_dispatch:
             self.setup_optimizers(trainer)
+
         self.precision_plugin.pre_dispatch()
 
+    def move_optimizer_state(self):
+        for opt in self.optimizers:
+            state = defaultdict(dict)
+            for p, v in opt.state.items():
+                state[p] = apply_to_collection(v, torch.Tensor, move_data_to_device, self.root_device)
+            opt.state = state
+
```

Comment:
```suggestion
    def _move_optimizer_state(self) -> None:
        for opt in self.optimizers:
            state = defaultdict(dict)
            for p, v in opt.state.items():
                state[p] = apply_to_collection(v, torch.Tensor, move_data_to_device, self.root_device)
            opt.state = state

```

mark as private unless we expect callers outside the accelerator class to need this. this wouldn't have any effect on sharded optimizers because that's handled afterward in the training type plugin pre_dispatch right?

Reply:
Done ! Yes, exactly.

Reply:
Actually.